src/sources/util/http.rs

```diff
@@ -49,6 +50,55 @@ impl fmt::Debug for RejectShuttingDown {
 }
 impl warp::reject::Reject for RejectShuttingDown {}
 
+#[derive(Deserialize, Serialize, Debug, Clone)]
+pub struct HttpSourceAuthConfig {
+    pub username: String,
+    pub password: String,
+}
+
+impl From<Option<&HttpSourceAuthConfig>> for HttpSourceAuth {
+    fn from(auth: Option<&HttpSourceAuthConfig>) -> Self {
+        let token = auth.map(|auth| {
+            let mut headers = HeaderMap::new();
+            headers.typed_insert(Authorization::basic(&auth.username, &auth.password));
+            headers
+                .get("authorization")
+                .unwrap()
+                .to_str()
+                .unwrap()
+                .to_owned()
+        });
+        HttpSourceAuth { token }
+    }
+}
+
+#[derive(Debug, Clone)]
+struct HttpSourceAuth {
+    pub token: Option<String>,
+}
+
+impl HttpSourceAuth {
+    pub fn is_valid(&self, header: Option<String>) -> Result<(), ErrorMessage> {
```

Comment:
```suggestion
    pub fn is_valid(&self, header: Option<&str>) -> Result<(), ErrorMessage> {
```

or 

```suggestion
    pub fn is_valid(&self, header: &Option<String>) -> Result<(), ErrorMessage> {
```

Otherwise you're consuming this memory and forcing allocations :)

Reply:
I'll change, but in our case, it's ok to have `Option<String>` because we do not need a clone object for this method.